server/metrics.go

```diff
@@ -199,4 +213,28 @@ func init() {
 	prometheus.MustRegister(apiConcurrencyGauge)
 	prometheus.MustRegister(forwardFailCounter)
 	prometheus.MustRegister(forwardTsoDuration)
+	prometheus.MustRegister(regionRequestCounter)
+}
+
+func incRegionRequestCounter(method string, header *pdpb.RequestHeader, err *pdpb.Error) {
+	if err == nil && rand.Intn(100) != 0 {
+		// sample 1% region requests to avoid high cardinality
+		return
+	}
```

Comment:
Instead of sampling with a specific probability, can we record the data into a counter and periodically flush it to Prometheus?

Reply:
Recording to the counter also requires calculation via a hash map, and we need to ensure it's thread-safe, which likewise introduces considerable overhead.